### PR TITLE
Add jump search

### PR DESCRIPTION
**Pull request by muthuannamalai12**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
I have added an algorithm for Jump Search in C
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2233 
(Write your answer here.)

### CODE REVIEW

File: C/search/JumpSearch.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	printf("\nEnter the element to be searched");
+	scanf("%d", &ele);
+
+	int flag = -99;
+
+	//Instead of checking consecutive elements one by one, we jump elements by a distance of sqrt(n).
+	int jmp = sqrt(n);
+	int i = 0, pos = jmp;
+	while (a[pos] <= ele && pos < n)	//We keep on jumping elements by a distance 'jmp' until the current array position becomes greater than the element to be searched.
```
Comment (atarax665) [inline C/search/JumpSearch.c:28]: Comment should be above the line of code not beside it.

File: C/search/JumpSearch.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	{
+		i = pos;
+		pos += jmp;
+		if (pos > n - 1)
+			flag = -1;
+	}
+	for (int j = i; j < pos; j++)	//Now we iterate from the last position 'i' where the a[i] <= ele, and iterate jmp(=sqrt(n) times in worst case.
```
Comment (atarax665) [inline C/search/JumpSearch.c:35]: Comment should be above the line of code not beside it.

File: C/search/JumpSearch.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	while (a[pos] <= ele && pos < n)	//We keep on jumping elements by a distance 'jmp' until the current array position becomes greater than the element to be searched.
+	{
+		i = pos;
+		pos += jmp;
+		if (pos > n - 1)
+			flag = -1;
+	}
+	for (int j = i; j < pos; j++)	//Now we iterate from the last position 'i' where the a[i] <= ele, and iterate jmp(=sqrt(n) times in worst case. 
+	{
+		if (a[j] == ele)	// If element is found we store in the variable flag, the final answer of the position where its found.
```
Comment (atarax665) [inline C/search/JumpSearch.c:37]: Comment should be above the line of code not beside it.

File: C/search/JumpSearch.c [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			flag = j;
+	}
+
+	if (flag >= 0)
+		printf("\n Element Found at %dth position", flag);
```
Comment (atarax665) [inline C/search/JumpSearch.c:42]: Use a better output message. Something similar to *Element found at index :*

File: C/search/JumpSearch.c [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	for (int i = 0; i < n; i++)
+		scanf("%d", &a[i]);
+
+	printf("\nEnter the element to be searched");
+	scanf("%d", &ele);
+
+	int flag = -99;
+
+	//Instead of checking consecutive elements one by one, we jump elements by a distance of sqrt(n).
+	int jmp = sqrt(n);
```
Comment (atarax665) [inline C/search/JumpSearch.c:26]: The ```main``` block should only have input-output statements. Make a new function and shift all the logic there.

File: C/search/JumpSearch.c [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+/*
+Sample Case:
+Example 1:
+Enter the size of array 5                                                                                                                                                                                                                                       
+Enter the sorted array elements 1 2 3 4 5                                                                                                                                                                                                                       
+Enter the element to be searched 3                                                                                                                                                                                                                             
+Element Found at 2th position  
+
+Example 2:
```
Comment (atarax665) [inline C/search/JumpSearch.c:55]: One sample I/O is enough.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
@muthuannamalai12 Are you working on this?

**Comment by muthuannamalai12** [comment]
> @muthuannamalai12 Are you working on this?

Yes I am working on it

**Comment by HarshCasper** [comment]
Closing this because of no updates

**Pull request closed without merging**